sql/src/main/java/io/crate/execution/engine/collect/stats/JobsLogs.java

```diff
@@ -115,10 +119,14 @@ public void logExecutionEnd(UUID jobId, @Nullable String errorMessage) {
         if (!isEnabled() || jobContext == null) {
             return;
         }
-        LogSink<JobContextLog> jobContextLogs = jobsLog.get();
-        JobContextLog contextLog = new JobContextLog(jobContext, errorMessage);
-        addToHistogram(contextLog);
-        jobContextLogs.add(contextLog);
+        jobsLogRWLock.readLock().lock();
+        try {
+            JobContextLog contextLog = new JobContextLog(jobContext, errorMessage);
+            addToHistogram(contextLog);
```

Comment:
I think in a couple of places the actions besides the `jobsLog.add` could be moved out of the lock

Reply:
right, changed it.